File: TypeScript/arrays-and-hashing/containsDuplicateBest.ts
class Solution {
    hasDuplicate(nums: number[]):boolean {
        const vistos = new Set<number>();
        for(const num of nums){
            if(vistos.has(num))
                return true;
            vistos.add(num);
        }
        return false;
    }
}

const nums: number[] = [2,2,1,3];
const ifHasDuplicate:boolean = new Solution().hasDuplicate(nums);
console.log(ifHasDuplicate);

File: Python/arrays-and-hashing/contains_duplicate_best.py
from typing import List

class Solution:

    def has_duplicate(self, nums: List[int]) -> bool:
        vistos = set()
        for num in nums:
            if num in vistos:
                return True
            vistos.add(num)
        return False
    
nums = [2,2,4,5]
if_has_duplicate = Solution().has_duplicate(nums)
print(if_has_duplicate)